src/test/java/PresenterTest.java

```diff
@@ -5,13 +5,19 @@
 import org.junit.Test;
 
 import java.io.IOException;
+import java.util.regex.Pattern;
 
 class FakeClientConnection extends ClientConnection {
-    public void setInput(String input){
+    public FakeClientConnection(FakeSocket fs) {
+        super(fs);
+    }
+
+    public void setInput(String input) {
```

Comment:
Why do we need to `setInput`? Is there a way we can construct the `FakeClientConnection` so that is in the desired state when it is instantiated?